fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledIteratorTest.java

```diff
@@ -580,7 +596,10 @@ void testLateCompleteFutures() throws Exception {
             try (FDBRecordContext context = openContext()) {
                 openSimpleRecordStore(context);
                 iterateAll = throttledIterator.iterateAll(recordStore.asBuilder());
+                commit(context);
             }
+            // "sleep" for 10 ms to allow futures to get scheduled
+            MoreAsyncUtil.delayedFuture(10, TimeUnit.MILLISECONDS).join();
```

Comment:
IIUC the thing you want to be testing here is that the `ThrottledIterator` is not interacting with the context used for the original store. To do that, I think you can remove the sleep by having the `timeHandler` do
```
CompletableFuture.runAsync(() -> semaphore.wait())
```
and release the semaphore after the context is closed.